Added key replacement method and refactored file tree for testing

ricdezen opened the pull request:
# Premise
The pull request regards two topics:
- The addition of a regex replacement method in `key_handler.py`, not much to say here
- The refactoring of the file tree in something that (should) not break.

### Structure
The structure is as follows
```
root
├── download.py
├── upload.py
├── run_all_tests.py
├── config.json
├── README.md
├── ...
├── otri
│   ├── downloader
│   │   ├── timeseries_downloader.py
│   │   └── ...
│   ├── importer
│   │   └── ...
│   └── ...
├── test
│   └── ...
├── docs
│   └── ...
└── data
    └── ...
```
### Imports
The imports always assume the working directory to be outside of the `otri` directory, just as it would be when using the module from outside. So the scripts like download.py and such, refer to directories/files starting from the root and have imports that look like:
```python
import otri.downloader.timeseries_downloader
from otri.database import *
```
Tests use absolute imports too, just like above, assuming we will always start them from the root directory.

Actual source files use relative imports:
```python
# Another module from the same package/subpackage
import .timeseries_downloader
# A module from another subpackage in the parent package
from ..database import database_adapter
```

I have tried downloading, uploading and running the very few tests I made, and everything seems to work.
Please note that the VSCode `python.terminal.runFilesInDir` option disrupts the logic of the file tree because it changes the working directory, which I remind should be the root.

Commit:
fix (key_handler.py) : removed useless test code in preparation for actual unit testing
fix : added missing __init__.py to root dir

Commit:
fix (*): moved all source files to an "otri" directory

Commit:
just the init.pys

Commit:
fix : missing data subdir in .gitignore

Commit:
Merge branch 'hotfix/filetree-refactor' of https://github.com/OTRI-Unipd/OTRI into feat/key_replacement

Commit:
feat : __init__.py

Commit:
fix (key_handler.py): fixed specs and formatting

Commit:
feat(run_all_tests.py): file to run all tests using pytest

Commit:
feat (key_handler_test.py): some tests for key_handler.py module

Commit:
fix : moved to root dir and changed to absolute imports

Commit:
fix : moved to root/docs/...

Commit:
fix : made imports relative

Commit:
fix: old files cleanup

Commit:
fix : fixed more paths or broke them idk anymore

Commit:
fix (run_all_tests.py) : using os instead of subprocess

Commit:
fix : codacy typo fix

Commit:
fix (key_handler.py) : changed type checks from type(obj) == type to isinstance(obj, type)

CremaLuca commented:
> Please note that the VSCode `python.terminal.runFilesInDir` option disrupts the logic of the file tree

So does that option have to be checked or not?

ricdezen commented:
Should NOT be checked

CremaLuca commented on otri/utils/key_handler.py:
```diff
@@ -97,16 +103,30 @@ def rename_deep(data, aliases : dict):
+        receiving the same treatment. It will return the original object (not a copy)
+        if no operation could be applied. See apply_deep(data, fun) for details.
+    '''
+    def replace_regex(string):
```
Nice and clean, really good work

CremaLuca commented on run_all_tests.py:
```diff
@@ -0,0 +1,3 @@
+import os
+
+os.system('cmd /c "python -m pytest"')
```
I'm interested: what does `cmd /c` do?

ricdezen replied:
`/c` is basically "exit after the process is done", not sure it would change much here

CremaLuca reviewed (approved):
Probably the perfect solution, easy and elegant, gj.

Commit:
fix (key_handler.py):
+ added missing return statement

fix (key_handler_test.py):
+ added a test for replace_deep

CremaLuca commented:
You might wanna add tests for the key replace method
